### PR TITLE
[Merged by Bors] - Support specifying Service type

### PR DESCRIPTION
**Pull request by sbernauer**
# Description

For https://github.com/stackabletech/issues/issues/360

*Please add a description here. This will become the commit message of the merge request later.*

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes
 
```[tasklist]
# Author
- [ ] Changes are OpenShift compatible
- [ ] CRD changes approved
- [ ] Helm chart can be installed and deployed operator works
- [ ] Integration tests passed (for non trivial changes)
```

```[tasklist]
# Reviewer
- [ ] Code contains useful comments
- [ ] (Integration-)Test cases added
- [ ] Documentation added or updated
- [ ] Changelog updated
- [ ] Cargo.toml only contains references to git tags (not specific commits or branches)
```

```[tasklist]
# Acceptance
- [ ] Feature Tracker has been updated
- [ ] Proper release label has been added
```

Once the review is done, comment `bors r+` (or `bors merge`) to merge. [Further information](https://bors.tech/documentation/getting-started/#reviewing-pull-requests)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -10,11 +10,13 @@ All notable changes to this project will be documented in this file.
 
 ### Changed
 
+- [BREAKING]: Moved top level config options (TLS, Authentication, Authorization etc.) to new top level field `clusterConfig` ([#400]).
+- [BREAKING]: Support specifying Service type by moving `serviceType` (which was an experimental feature) to `clusterConfig.listenerClass`.
+  This enables us to later switch non-breaking to using `ListenerClasses` for the exposition of Services ([#406]).
```
Comment (maltesander) [inline CHANGELOG.md:15]:
```suggestion
  This enables us to later switch non-breaking to using `ListenerClasses` for the exposure of Services ([#406]).
```

**Review by maltesander — APPROVED** [review]
LGTM if we agreed on the listener_class names :)

**Comment by sbernauer** [comment]
We chatted about this in #product-listener and decided to stick with the current names

**Comment by sbernauer** [comment]
Started test https://ci.stackable.tech/view/02%20Operator%20Tests%20(custom)/job/trino-operator-it-custom/64/

**Comment by sbernauer** [comment]
Test failed, but I think it's normal flakiness

**Comment by sbernauer** [comment]
Started https://ci.stackable.tech/view/02%20Operator%20Tests%20(custom)/job/trino-operator-it-custom/65/

**Comment by sbernauer** [comment]
bors r+
Test passed

**Pull request closed without merging**